python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -324,24 +353,23 @@ def build_failed_and_in_progress_partition_subset(
     asset_key: AssetKey,
     partitions_def: PartitionsDefinition,
     dynamic_partitions_store: DynamicPartitionsStore,
-    failed_partitions_subset: Optional[PartitionsSubset] = None,
+    last_planned_materialization_storage_id: int,
+    failed_partitions: Optional[Set[str]] = None,
     after_storage_id: Optional[int] = None,
 ) -> Tuple[PartitionsSubset, PartitionsSubset, Optional[int]]:
-    failed_partitions: Set[str] = (
-        set(failed_partitions_subset.get_partition_keys()) if failed_partitions_subset else set()
-    )
+    failed_partitions = failed_partitions or set()
     in_progress_partitions: Set[str] = set()
-    if failed_partitions:
-        # These partitions were cached as having been failed.  If they have since been materialized,
-        # then we can remove them from the set of failed partitions.
-        materialized_partitions = instance.event_log_storage.get_materialized_partitions(
-            asset_key, after_cursor=after_storage_id
-        )
-        failed_partitions.difference_update(materialized_partitions)
 
-    incomplete_materializations = instance.event_log_storage.get_latest_asset_partition_materialization_attempts_without_materializations(
-        asset_key, after_storage_id=after_storage_id
-    )
+    incomplete_materializations = {}
+
+    # Fetch incomplete materializations if there have been any planned materializations since the
+    # cursor
+    if last_planned_materialization_storage_id and (
+        not after_storage_id or last_planned_materialization_storage_id > after_storage_id
+    ):
```

Comment:
now we only do this query if we have reason to believe there has been a new planned materialization since the cursor